clang/lib/Sema/SemaBounds.cpp

```diff
@@ -614,6 +653,37 @@ namespace {
       DumpEqualExpr(OS, State.G);
     }
 
+    void DumpBoundsContext(raw_ostream &OS, BoundsContextTy UC) {
+      if (UC.empty())
+        OS << "{ }\n";
+      else {
+        // The keys in an llvm::DenseMap are unordered.  Create a set of
+        // variable declarations in the context ordered first by name,
+        // then by location in order to guarantee a deterministic output
+        // so that printing the bounds context can be tested.
+        std::vector<DeclaratorDecl *> OrderedDecls;
+        for (auto Pair : UC)
+          OrderedDecls.push_back(Pair.first);
+        llvm::sort(OrderedDecls.begin(), OrderedDecls.end(),
+             [] (DeclaratorDecl *A, DeclaratorDecl *B) {
+               if (A->getNameAsString() == B->getNameAsString())
```

Comment:
You could write this comparator more succinctly as:
`
return std::tie(A->getNameAsString(), A->getLocation()) <
           std::tie(B->getNameAsString(), B->getLocation())
`

Reply:
After I made this change, there were build warnings related to `A->getNameAsString()`, etc. not being lvalues so I've reverted it at least for now.